### ISSUE TITLE
Support Angular/Material v10

### ISSUE THREAD

**Issue by qortex** [swimlane/ngx-charts#1467]
Migrating to Angular/Material v10 throws:
```
Package "@swimlane/ngx-charts" has an incompatible peer dependency to "@angular/cdk" (requires "^9.2.0", would install "10.0.0").
```

v10 being a small update, it's likely `ngx-charts` will be compatible as is.

**Comment by blogcraft** [comment]
This project feels abandoned more and more. 😔

**Comment by OfirArbel** [comment]
follow

**Comment by qortex** [comment]
> follow

(Fyi, the "subscribe" button on the right allows to follow the issue without posting & notifying all following people.)

**Comment by linjie997** [comment]
For the instance, you can update using `--force` flag with ng update. I can confirm the charts works with Angular 10.

**Comment by linjie997** [comment]
I created a pull request with updated dependencies, hopefully it gets reviewed soon.

**Comment by bellamira** [comment]
Will this be available on NPM soon? The last release I see was April. Thank you!

> I created a pull request with updated dependencies, hopefully it gets reviewed soon.
> 
> Edit: Got merged :)

**Comment by linjie997** [comment]
> Will this be available on NPM soon? The last release I see was April. Thank you!
> 
> > I created a pull request with updated dependencies, hopefully it gets reviewed soon.
> > 
> > Edit: Got merged :)
> 
> 

Hopefully soon, that's up to swimlane team.

**Comment by trickeyone** [comment]
Yeah, thankfully there isn't typically much needed to upgrade from 9 to 10. So, hopefully it gets merged/released soon

**Comment by PPInfy** [comment]
Can you release it in your branch @linjie997. I need it to test the same.

**Comment by miggymigz** [comment]
Hey, can you also do some changes so that the latest version of angular doesn't show warnings like these?

```bash
Warning: Unable to fully load /home/miguelvictor/Projects/mimic-lstm-angular/node_modules/@angular/material/sidenav/drawer.d.ts for source-map flattening: Circular source file mapping dependency: /home/miguelvictor/Projects/mimic-lstm-angular/node_modules/@angular/material/sidenav/drawer.d.ts.map -> /home/miguelvictor/Projects/mimic-lstm-angular/node_modules/@angular/material/sidenav/drawer.d.ts.map
```

**Comment by linjie997** [comment]
> Hey, can you also do some changes so that the latest version of angular doesn't show warnings like these?
> 
> ```shell
> Warning: Unable to fully load /home/miguelvictor/Projects/mimic-lstm-angular/node_modules/@swimlane/ngx-charts/swimlane-ngx-charts.d.ts for source-map flattening: Circular source file mapping dependency: /home/miguelvictor/Projects/mimic-lstm-angular/node_modules/@swimlane/ngx-charts/swimlane-ngx-charts.d.ts.map -> /home/miguelvictor/Projects/mimic-lstm-angular/node_modules/@swimlane/ngx-charts/swimlane-ngx-charts.d.ts.map
> ```

I removed @angular and @swimlane folder from node_modules, package-lock.json, run `npm install --save` and the warnings disappeared.

**Comment by lukeatron** [comment]
> > Will this be available on NPM soon? The last release I see was April. Thank you!
> > > I created a pull request with updated dependencies, hopefully it gets reviewed soon.
> > > Edit: Got merged :)
> 
> Hopefully soon, that's up to swimlane team.

It's been two weeks.  What's the holdup?

**Comment by trickeyone** [comment]
@marjan-georgiev Would it be possible to get an ETA on when a release will be made with the #1478 support for Angular 10?

**Comment by Tea-Ice** [comment]
I think it is pretty clear that the team no longer cares, not even enough to make a release of already merged in code.

**Comment by trickeyone** [comment]
Just a heads-up to everybody else waiting on this. I did notice that some charts stopped working when I migrated to Angular v10 without this fix. Looks like it's some issues with the version of the various d3 components that ngx-charts utilizes. I was able to get the charts back to working by explicitly requiring updates version of the packages:

```
{
  "dependencies": {
    ...
    "d3-array": "^2.5.0",
    "d3-brush": "^1.1.5",
    "d3-color": "^1.4.1",
    "d3-format": "^1.4.4",
    "d3-hierarchy": "^1.1.9",
    "d3-interpolate": "^1.4.0",
    "d3-scale": "^3.2.1",
    "d3-selection": "^1.4.2",
    "d3-shape": "^1.3.7",
    "d3-time-format": "^2.2.3",
    "d3-transition": "^1.3.2",
    ...
  }
}
```

**Comment by jpduckwo** [comment]
Hanging out for the update... any news? 🤞

@trickeyone sounds a bit odd, did you previously have d3 packages installed explicitly? I would think that ngx-charts would pull in the d3 dependencies it works with, the angular version shouldn't have really much to do with it. It could be something specific to your project, if someone else can confirm or you can confirm its not specific then it might be worth raising a separate issue.

**Comment by trickeyone** [comment]
@jpduckwo I had the main d3 library installed. But then I started getting errors after upgrading to Angular v10. The error was coming from inside the d3-selection package, which I didn't have explicitly required. I had the d3 package required simply because a previous version of ngx-charts, I believe, had that as the install instructions. So, I'm guessing that they used to just tie into the full d3 package, instead of individual components. After I explicitly required a newer version of the packages, my charts started working. If your charts are still working after upgrading to Angular v10, then idk.

After I removed the d3 package, I did a prune, deleted `node_modules`, etc. The charts would always error.

**Comment by lukeatron** [comment]
Any day now some one will work the energy to slide their mouse over to the publish button.  Yup, completely reasonable that it take a month to do.

**Comment by blogcraft** [comment]
I just got to terms with this. I took the painful decision to remove this library from all our projects and move to amCharts.

**Comment by marjan-georgiev** [comment]
Thanks for the patience everyone, and apologies for the longer wait. Published `16.0.0`
